### ISSUE TITLE
Locking on primitives and/or Longs

### ISSUE THREAD

**Issue by borkdude** [juji-io/datalevin#5]
I stumbled upon an issue while trying to compile this project with GraalVM native-image: https://github.com/borkdude/datalevin-native

Just sharing a conversation I had with Alex Miller on Slack about it. 

``` 
alexmiller: You can't lock on a primitive long - it has to be an object
That code doesn't make sense
borkdude yeah, I figured the same. it comes from here:
https://github.com/juji-io/datalevin/blob/0588dde496b18ced06fed036db3f969c35c69ef8/src/datalevin/storage.clj#L226 
alexmiller yeah, that's bad
3:14 PM
borkdude so it does not work, but it doesn't fail in the JVM either?
3:25 PM
alexmiller my guess would be that in some cases the primitive long is getting up cast. that's the only way it could work (but locking on a Long value is still a bad idea). this code is just wrong.
3:27 PM
the Java numeric value objects can be cached around 0 and that means the scope of your Long sharing can escape your own code (similarly, you should never lock on a String instance which might be interned) (edited) 
3:28 PM
borkdude :thumbsup: I'll make an issue with this conversation in their repo
alexmiller it's best to make an explicit lock Object
```

**Comment by huahaiy** [comment]
Fixed in 0.2.4

**Comment by huahaiy** [comment]
kk
